testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/LoginTest.java

```diff
@@ -894,6 +893,28 @@ public void openLoginFormAfterExpiredCode() throws Exception {
         events.expectLogin().detail(Details.USERNAME, "test-user@localhost").assertEvent();
     }
 
+    @Test
+    public void testAuthenticationSessionExpiresEarlyAfterAuthentication() throws Exception {
+        // Open login form and refresh right after. This simulates creating another "tab" in rootAuthenticationSession
+        oauth.openLoginForm();
+        driver.navigate().refresh();
+
+        // Assert authenticationSession in cache with 2 tabs
+        String authSessionId = driver.manage().getCookieNamed(AuthenticationSessionManager.AUTH_SESSION_ID).getValue();
+        Assert.assertEquals((Integer) 2, getTestingClient().testing().getAuthenticationSessionTabsCount("test", authSessionId));
+
+        loginPage.login("test-user@localhost", "password");
+        appPage.assertCurrent();
+
+        // authentication session should still exists with remaining browser tab
+        Assert.assertEquals((Integer) 1, getTestingClient().testing().getAuthenticationSessionTabsCount("test", authSessionId));
+
+        // authentication session should be expired after 1 minute
+        setTimeOffset(300);
+        Assert.assertEquals((Integer) (-1), getTestingClient().testing().getAuthenticationSessionTabsCount("test", authSessionId));
```

Comment:
Looks a but odd to see -1 here if the last remaining auth session was removed. I'd expect this to be 0.

Reply:
Yeah it looks weird, given the method name. Would it be sufficient return also 0 when there in no root auth session?